### ISSUE TITLE
Unable to install on ARM64

### ISSUE THREAD

**Issue by jimmad123** [KatharaFramework/Kathara#104]
**Describe the bug**
I am trying to install Kathara on a Raspberry Pi 3B+ (running a 64-bit quad-core ARMv8 processor). I have followed instructions to install and verify that Docker is working and, when it comes to the final step of the Kathara installation instructions for Linux (the RPi is running Ubuntu 18.04) it says "Unable to read package kathara".

**To Reproduce**
1. Use a device with an ARM64 processor and Ubuntu
2. Follow the Kathara installation instructions (https://github.com/KatharaFramework/Kathara/wiki/Linux)
3. At the last step ("sudo apt install kathara") you will see the error

**Expected behavior**
I expected the Kathara install to be successful on an ARM64 processor as it is listed within the InRelease file

**Screenshots**
![image](https://user-images.githubusercontent.com/44371017/95894137-94e5df80-0d80-11eb-876d-eec4b6da0bf8.png)

<details>
<summary>"check" Command Output</summary>
kathara: command not found
</details>

**Comment by Skazza94** [comment]
Hi, as stated [here](https://github.com/KatharaFramework/Kathara/blob/master/scripts/Linux/debian/control#L16), we only compile and deploy Kathará for amd64 on Launchpad.

However, if you want, you can try to compile it for arm64, and if it works we can add this architecture to the build pipeline.

To do so, pull the latest Kathará code and navigate `scripts/Linux`. 
In this folder type `make docker-unsigned`. This will:

- Create a proper Docker image for the build environment
- Run a Docker container with the image built above
- Compile Kathara into a binary
- Create a unsigned .deb package
- Automatically remove the Docker container

The compiled `.deb` package is in the `Output` folder. Install it typing;
```
sudo apt install ./<FILE_NAME>.deb
```

Let us know,
Mariano.

**Comment by jimmad123** [comment]
Hi,

I have spent a great deal of time trying to compile Kathará on my Raspberry Pi, but run into many errors after running `make docker-unsigned`. It seems as though the Debian container that is created with that command is out-of-date, as there are certain packages that depend on newer builds of Python that can be installed in the container.

Just to make sure that it wasn't an ARM issue, I have tried running the unsigned build on amd64 and come across exactly the same errors.

Is this a known issue when trying to run a build?

**Comment by Skazza94** [comment]
Hi @jimmad123,
could you please share the errors so we can furtherly investigate?

Thanks.

**Comment by jimmad123** [comment]
Hi @Skazzino,

The initial error comes when the command `gem install nokogiri mustache kramdown rubocop sinatra rack rake test-unit`is run within the Docker container. 

This error arises as `rubocop` requires a higher Ruby version than is available in Debian 9 (the OS for the Docker build environment container) as can be seen below:

![RubocopRubyError](https://user-images.githubusercontent.com/44371017/96934966-0f1e0e80-14bb-11eb-929c-69f4b0e1639e.png)

The gem command then installs the remaining packages, but once that's finished the process exits as such:

![DockerBuildEnvError](https://user-images.githubusercontent.com/44371017/96935095-50162300-14bb-11eb-8c7c-d2f911b0e98d.png)

There are various similar errors that occur once that is finished and I managed to fix most of the package errors (as far as I know) by modifying the Dockerfile so that it installs `Rubocop v0.81`, as well as installing `kramdown-parser-gfm` as that threw an error later in the build due to "no way to parse kramdown" or something similar.

Ultimately, my Dockerfile gem commands ended up looking like this:

![DockerfileSection](https://user-images.githubusercontent.com/44371017/96935514-1eea2280-14bc-11eb-865b-f50c6354c370.png)

The next batch of errors come from the Makefile. When the `copy-debian-folder` task is executed from said file, it appears that the `debian` folder is copied into the `Output/kathara-$(VERSION)/` directory, however that is not created at any point before the copy command is executed:

![CopyError](https://user-images.githubusercontent.com/44371017/96936402-e21f2b00-14bd-11eb-81c5-969a13d27e05.png)

As such, I added a `mkdir` command before the copy:

![CopyDebianFolder](https://user-images.githubusercontent.com/44371017/96936054-42619d00-14bd-11eb-9813-9e5e85ebc866.png)

The next error I found is in the `download-pip` task. Python is downloaded as per the `requirements.txt` file, but the path given to that file is incorrect:

![RequirementsFileError](https://user-images.githubusercontent.com/44371017/96936554-34f8e280-14be-11eb-94c1-f1e5d2fd5267.png)

This was corrected by manually moving the `requirements.txt` file from the `Kathara/src/` directory to the `scripts/Linux/` directory, followed by a change to the Makefile's `download-pip` task to copy the file into the expected directory:

![RequirementsCopy](https://user-images.githubusercontent.com/44371017/96937320-c583f280-14bf-11eb-95e8-243184bb4be5.png)

This seems to fix the Dockerfile and Makefile errors that I encountered. The next step was to deal with errors from the `debuild -us -uc` command. 

This initially complained that there was no "appropriate original tar file or .orig directory in the parent directory;", so I manually downloaded this file from your PPA Launchpad and placed it in the parent directory by editing the `build-deb-unsigned` task in the Makefile. The next error comes when the `kathara.py` file cannot be found in the `Output/kathara-$(VERSION)/` directory, so that was fixed in the same Makefile task as above:

![OrigKatharaPyCopy](https://user-images.githubusercontent.com/44371017/96938401-34624b00-14c2-11eb-9228-2f2261da7cdf.png)

That is the end of the errors that I get while building on amd64. I hope that helps with the errors (if you get them as well). 

The next error I get when trying to build on arm64 I haven't been able to fix yet. It is as follows:

![PiDHError](https://user-images.githubusercontent.com/44371017/96940254-26fb8f80-14c7-11eb-995a-b5578bedc432.png)

If you have any ideas at all as to how I can deal with that error, please let me know! :)

Thank you

**Comment by jimmad123** [comment]
@Skazzino just a short update, I have realised that the error I was seeing on my Pi in the last post was due to incorrect `Architecture` specified in the `control` file. Instead, there is now an error with `collect2`, as seen here:

![PiCollectError](https://user-images.githubusercontent.com/44371017/96941239-d3d70c00-14c9-11eb-8985-2f1de3719454.png)

I imagine you've never come across this error whilst building Kathara for amd64 but, on the off chance you know how to fix this, I would really appreciate assistance.

Thanks again

**Comment by Skazza94** [comment]
Hi @jimmad123 and thanks for you time invested in doing this.
First of all, the `Makefile` of the `master` branch seems to be buggy. Please, use the `new_megalos` branch (it'll be the next release) and use that build pipeline, we changed the building stuff a little bit.

Remember to change the `Architecture` into `scripts/Linux/debian/control` and let us know

**Comment by Skazza94** [comment]
Hi @jimmad123,
just as reminder if you're still interested in compiling Kathará on ARM, we merged `new_megalos` into the `master` branch.

Thanks,
Mariano.

**Comment by Skazza94** [comment]
Since the last reply was 21 days ago I'm considering the issue as solved or no longer relevant.

If you need further support, you can open another issue.

Thanks!
